components/ContactForm/index.tsx

```diff
@@ -67,13 +66,47 @@ export const ContactForm = () => {
     return isValid;
   };
 
-  const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
+  const handleSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
     e.preventDefault();
-    formIsValid() ? console.log(formState) : console.log("Form is invalid");
+
+    if (formIsValid()) {
+      const target = e.target as typeof e.target & {
+        name: { value: string };
+        email: { value: string };
+        message: { value: string };
+      };
+
+      const data = {
+        name: target.name.value,
+        email: target.email.value,
+        message: target.message.value,
+      };
+
+      const JSONdata = JSON.stringify(data);
+      const endpoint = "/api/contact-us";
+      const options = {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSONdata,
+      };
+
+      const response = await fetch(endpoint, options);
+      const result = await response.json();
```

Comment:
You should abstract this away into a single async `sendContactDetails(details: TDetails) => Promise<string>` function in a separate file (you can call it `requests.ts` for example).

The function should conceal the endpoint, options etc., so that in the React component you're just declaratively calling it, instead of imperatively writing out all options.